src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -341,18 +341,11 @@ private bool CanMapFromSecondaryBufferToPrimaryBuffer(Workspace workspace, Docum
 
         private IDisposable OpenNewDocumentStateScope(OptionSet options)
         {
-            if (!options.GetOption(NavigationOptions.PreferProvisionalTab))
-            {
-                return null;
-            }
-
-            var state = __VSNEWDOCUMENTSTATE.NDS_Provisional;
+            var state = options.GetOption(NavigationOptions.PreferProvisionalTab)
+                ? __VSNEWDOCUMENTSTATE.NDS_Provisional
+                : __VSNEWDOCUMENTSTATE.NDS_Permanent;
```

Comment:
this seems like a slight change in behavior, but I'm not sure it matters

Previously, it seems like permanent tabs will always be activated (or not not activated), but now we can have a permanent tab created but not activated.  Does that case ever exist?

Reply:
It's definitely a change, the reasoning is simply to copy the platform code [here](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform?path=%2Fsrc%2FEditor%2FVisualStudio%2FImpl%2FUtilities.cs&version=GBmaster&line=160&lineEnd=161&lineStartColumn=1&lineEndColumn=1&lineStyle=plain), and because my reading of [the documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.__vsnewdocumentstate?view=visualstudiosdk-2019) doesn't tie the `NoActivate` flag to provisional in any way.

I don't know if there is a scenario now where permanent tabs should be not activated, but it's possible something, now or in the future, passes a `TableEntryNavigateEventArgs` with `isPreview` and `shouldActivate` both set to false.

Reply:
This aligns with my understand of how we should ingest the `shouldActivate` from the platform. Before we only cared about activation for provisional because there were no cases where we wouldn't activate. The platform has changed that assumption though, and we're going to honor that flag. It is a little odd that they tell us what they expect us to tell them, but there's more nuances there.